### ISSUE TITLE
Proposal: Object Storage metrics refactor

### ISSUE THREAD

**Issue by douglascamata** [thanos-io/thanos#6224]
## Context

Thanos' Object Store has been extracted to [its own repository](https://github.com/thanos-io/objstore) to serve a higher purpose. Accordingly, now it exports its metrics under the `objstore_` prefix instead of `thanos_`. 

## Problem

This is a breaking change to everyone that already has dashboards and alerts on metrics from the objstore. 

## Proposed Solution

As a first step and potentially temporary solution #6152 was created to wrap the objstore metrics with the `thanos_` prefix, avoiding the breaking change.

As suggested by @bwplotka, there are many good reasons to **not wrap** the metrics and adopt the new name:

> It's all metrics about buckets operations, so semantically it makes sense, no matter who performs those bucket operations (job or any other target label will tell you). Again it allows objstore_ ALERTs and dashboards to be state of the art, reusable by anybody using objstore component.

We can follow up with one of the following:

* A deprecation period, in which the metrics are duplicated in both places for a few releases. After these "few releases", the old metric name is completely dropped.
* A command-line argument is added to switch between new and old objstore metrics.

⚠️ **Important**: this is not an extensive list of all the possible solutions. If anyone has a different idea that could work well, feel free to propose it.

### Recommended solution

My recommended solution is using duplication + deprecation period, as it allows to simplify the code later. The command-line argument option is more complicated in terms of deprecation and removal.

**Comment by hackeramitkumar** [comment]
@douglascamata Yeah I also agreed by your solution. It's really make sense. As I am new to this project and want to contribute. Would like to contribute to it.

**Comment by douglascamata** [comment]
@hackeramitkumar sure thing. Feel free to start working on it and open a PR whenever you are ready. Don't hesitate to try to reach for the maintainers here or in the CNCF slack #thanos-dev  channel 👍
